src/main/scala/com/microsoft/partnercatalyst/fortis/spark/Pipeline.scala

```diff
@@ -0,0 +1,100 @@
+import org.apache.spark.streaming.StreamingContext
+import org.apache.spark.streaming.dstream.DStream
+
+object Pipeline {
+  def apply[T: TypeTag](name: String, analyzer: Analyzer[T], ssc: StreamingContext, streamProvider: StreamProvider, configurationManager: ConfigurationManager): Option[DStream[FortisEvent]] = {
+    val configs = configurationManager.fetchStreamConfiguration(name)
+    val sourceStream = streamProvider.buildStream[T](ssc, configs)
+
+    sourceStream.map(_.transform(rdd => {
+
+      // TODO: populate these local variables with values from global update/broadcast wrappers,
+      // which will use the ConfigurationManager (cassandra) to update themselves and (where applicable)
+      // broadcast changes across the Spark cluster.
+
+      val geofence = Geofence(north = 49.6185146245, west = -124.9578052195, south = 46.8691952854, east = -121.0945042053)
+      val modelsProvider = new ZipModelsProvider(
+        language => s"https://fortiscentral.blob.core.windows.net/opener/opener-$language.zip",
+        Settings.modelsDir)
+
+      val featureServiceClient = new FeatureServiceClient(Settings.featureServiceHost)
+      val locationsExtractorFactory = new LocationsExtractorFactory(featureServiceClient, geofence).buildLookup()
+      val locationFetcher = locationsExtractorFactory.fetch _
+      val keywordExtractor = new KeywordExtractor(List("Ariana"))
+      val imageAnalyzer = new ImageAnalyzer(ImageAnalysisAuth(Settings.oxfordVisionToken), featureServiceClient)
+      val languageDetector = new LanguageDetector(LanguageDetectorAuth(Settings.oxfordLanguageToken))
+      val sentimentDetectorAuth = SentimentDetectorAuth(Settings.oxfordLanguageToken)
+      val supportedLanguages = Set("en", "fr", "de")
+
+      def convertToSchema(original: T): ExtendedFortisEvent[T] = {
+        val message = analyzer.toSchema(original, locationFetcher, imageAnalyzer)
+        ExtendedFortisEvent(message, Analysis())
+      }
+
+      def addLanguage(event: ExtendedFortisEvent[T]): ExtendedFortisEvent[T] = {
+        val language = analyzer.detectLanguage(event.details, languageDetector)
+        event.copy(analysis = Analysis(language = language))
+      }
+
+      def isLanguageSupported(analysis: Analysis): Boolean = {
+        analysis.language match {
+          case None => false
+          case Some(language) => supportedLanguages.contains(language)
+        }
+      }
+
+      def addKeywords(event: ExtendedFortisEvent[T]): ExtendedFortisEvent[T] = {
+        val keywords = analyzer.extractKeywords(event.details, keywordExtractor)
+        event.copy(analysis = event.analysis.copy(keywords = keywords))
+      }
+
+      def hasKeywords(analysis: Analysis): Boolean = {
+        analysis.keywords.nonEmpty
+      }
+
+      def addEntities(event: ExtendedFortisEvent[T]): ExtendedFortisEvent[T] = {
+        val entities = analyzer.extractEntities(event.details, new PeopleRecognizer(modelsProvider, event.analysis.language))
```

Comment:
It's expensive to instantiate this object (needs to download multi-megabyte model files and then load them into memory) so we should avoid doing so. The previous version with one static recognizer that does not take the language in the constructor avoids this issue.